notes.md

```diff
@@ -0,0 +1,570 @@
+# balenaOS in container for .local BoB usage
+Issue about running balenaOS in container and connect to .local bob e.g.: 55ccb57ae329e48ecf6b9ec7d42651a7.bob.local
+
+I wanted to use balenaOS in container for testing on my local bob. I wanted to be able to scale up multiple device easily without the need for a virtual machine setup.
+
+The repository: https://github.com/balena-os/balenaos-in-container is working out of the box for remotely hosted balena instances, like production or staging.
+
+When I tried the setup with my local BoB instance it failed to connect.
+First I setup a static (/etc/hosts) name resolution which worked from shell level. In the supervisor itself I faced the issue that the supervisor isn't able to resolve the .local domain name of the BoB.
+
+Then I started to understand what happens in the supervisor and the supervisor has a hard coded switch internally that is not using the os / node runtime DNSlookup functionality but overwrites the DNSlookup method to implement a nodejs level mDNS resolution. It's basically executing the mDNS broadcast in nodejs.
+
+Starting to work on a fix I setup myself with the supervisor Docker-in-Docker development setup to test the implementations. This setup didn't fail to connect to my .local BoB instance.
+
+The difference between the supervisor DinD and the balenaOSinContainer setup is the resinOS version.
+
+Finally I found out, that the resinOS 2.48.0_rev3-intel-nuc image works with .local bob and static name resolution. The 2.68... version used prior was failing to resolve the domain names.
+
+It's failing already in resinOS version 2.50, thus the linux Kernel 4 to 5 update isn't the root cause. Moreover, the latest supervisor code is working in the old resinOS 2.48 as this is used for the development and debugging.
```

Comment:
> linux Kernel 4 to 5 update

remember that this is a container, so we're not running the kernel here. I don't think anything that the balena-intel layer changed would be the cause for this

> latest supervisor code is working in the old resinOS 2.48

if the supervisor implements the mdns broadcast itself in nodejs, it should be independent of the mdns implementation in balenaOS, correct. can we repro this somehow using a simpler application that the SV, a user container that does mdns resolution in the same way should also break in the same way...